FirebaseDatabase/Sources/Core/FRepo.m

```diff
@@ -565,25 +586,33 @@ - (void)getData:(FIRDatabaseQuery *)query
                }
                [self.eventRaiser raiseCallback:^{
                  block(nil, [[FIRDataSnapshot alloc] initWithRef:query.ref
-                                                     indexedNode:node]);
+                                                     indexedNode:persisted]);
                }];
            } else {
-               node = [FSnapshotUtilities nodeFrom:data];
-               [self.eventRaiser
-                   raiseEvents:[self.serverSyncTree
-                                   applyServerOverwriteAtPath:[query path]
-                                                      newData:node]];
+               NSNumber *tagId =
+                   ![querySpec loadsAllData]
+                       ? [[self serverSyncTree] trackTagForQuery:querySpec]
+                       : nil;
+               // Set up a short-lived FKeepSyncedEventRegistration to populate
+               // cache with the data update.
+               [self keepQuery:querySpec synced:YES skipListen:YES];
+               NSArray *events = [self processDataUpdateForPath:query.path
```

Comment:
Could we consider adding ObjC generics to new and modified code? As someone new to the codebase it's difficult to guess the exact types held in these collections without digging into the underlying implementation.